src/components/SpectralProfiler/SpectralProfilerComponent.tsx

```diff
@@ -74,7 +122,8 @@ export class SpectralProfilerComponent extends React.Component<WidgetProps> {
         if (frame.regionSet) {
             const region = frame.regionSet.regions.find(r => r.regionId === regionId);
             if (region) {
-                coordinateData = this.profileStore.getProfile(this.widgetStore.coordinate, region.isClosedRegion ? this.widgetStore.statsType : CARTA.StatsType.Sum);
+                // Qi, calculate PI or PA if coordinate is PI or PA
+                coordinateData = this.calculate(this.widgetStore.coordinate, region.isClosedRegion ? this.widgetStore.statsType : CARTA.StatsType.Sum);
```

Comment:
could we call this function something a little less generic than `calculate`? Perhaps `calculateCompositeProfile`?